WordPress/gutenberg
# Block Library: Add home link block

This adds a new dynamic block that renders a link, backed by `home_url()`. Fixes https://github.com/WordPress/gutenberg/issues/20542

We're keeping this block simple to start. There's no url to edit, and subnesting of navigation links is not implemented.

We're also deferring linking to Reader Options (to modify home page settings) in the block settings panel for a follow up PR. We'll need a generalized approach here to allow folks to override these links with a new filter or slot/fill pattern.

There's a decent amount of copy pasted code needed for color/font support from the navigation parent. I do think this can and should be reduced.

For now, I opted to choose a similar implementation as navigation-link for now, while we consider options for what Navigation Block markup might look like. (See https://github.com/WordPress/gutenberg/issues/29036 )

https://user-images.githubusercontent.com/1270189/117212717-88be7f80-adaf-11eb-8f9a-9c9712321ed8.mp4

### Testing Instructions

- In a site or post editor, add a navigation block
- Clicking on the inserter provides a Home Link block option
- Verify that we cannot add a home link block outside of a navigation block parent
- Try to set some custom font sizes/colors on the navigation block parent. Home link styling should appropriately be affected by these changes.

~~Current work here is a very early WIP, but before I get too deep I have a few questions. We currently favor implementing this as a new block, rather than a navigation link variation:~~

- ~~Should we support this only in Navigation? Or are there use cases for adding this outside of the Navigation block, similar to PageList?~~ A: Only in navigation to start
- ~~Should we support child nesting of NavigationLinks inside of a home block? This will likely duplicate logic and complicate the block.~~ A: Keep it shallow
- ~~Naming: is `core/home` okay, or do we prefer something else?~~ A: `core/home-link`
- ~~The user shouldn't have to manually update the url, as such do we still need to display a link control UI? The only piece they may want to toggle is if the link opens in a new tab.~~ A: No need for link UI

Folks are welcome to jump directly onto this branch if you'd like to collaborate. Just leave a comment or DM me so I'll know to a give heads up before rebasing.
